src/Aquifer.Public.API/Endpoints/Bibles/List/Endpoint.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Aquifer.Data;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.Public.API.Endpoints.Bibles.List;
+
+public class Endpoint(AquiferDbContext dbContext) : Endpoint<Request, IReadOnlyList<Response>>
+{
+    private readonly AquiferDbContext _dbContext = dbContext;
+
+    public override void Configure()
+    {
+        Get("/bibles");
+        Description(d => d.ProducesProblemFE());
+        Summary(s =>
+        {
+            s.Summary = "Get a list of Bibles.";
+            s.Description =
+                "For a given optional language ID, returns the Bibles in the system including language information, abbreviation codes, whether or not the Bible supports a Greek alignment, and license information.";
+        });
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
+    {
+        var bibles = await _dbContext.Bibles
+            .Where(b =>
+                b.Enabled &&
+                !b.RestrictedLicense &&
+                (!request.LanguageId.HasValue || b.LanguageId == request.LanguageId) &&
+                (request.LanguageCode == null || b.Language.ISO6393Code == request.LanguageCode) &&
+                (!request.IsLanguageDefault.HasValue || b.LanguageDefault == request.IsLanguageDefault) &&
+                (!request.HasGreekAlignment.HasValue || b.GreekAlignment == request.HasGreekAlignment) &&
+                (!request.HasAudio.HasValue || b.BibleBookContents.Any(bbc => bbc.AudioUrls != null) == request.HasAudio))
+            .Select(bible => new Response
+            {
+                Name = bible.Name,
+                Abbreviation = bible.Abbreviation,
+                Id = bible.Id,
+                SerializedLicenseInfo = bible.LicenseInfo,
+                LanguageId = bible.LanguageId,
+                IsLanguageDefault = bible.LanguageDefault,
+                HasGreekAlignment = bible.GreekAlignment,
+                HasAudio = bible.BibleBookContents.Any(bbc => bbc.AudioUrls != null),
+            })
+            .ToListAsync(ct);
+
+        await SendOkAsync(bibles, ct);
```

Comment:
This is fine and we're not consistent, but as an FYI, you could just say `Response = ...` above (instead of `var bibles = ...`) and remove this line

Reply:
I copied [the internal API](https://github.com/BiblioNexusStudio/aquifer-server/blob/81626d70a745af94b4eada04109100b5169ca973/src/Aquifer.API/Endpoints/Bibles/List/Endpoint.cs#L34).

I can update both new endpoints to set the response property if that's best practice.

Reply:
I did some more research here to understand what FastEndpoints is doing under the hood.

My question: Does setting the `Response` property directly [per these docs](https://fast-endpoints.com/docs/get-started#sending-responses) in `HandleAsync()` have the semantic equivalence to calling `await SendOkAsync(myResponseObject, cancellationToken);` inside `HandleAsync()`?

After some source code digging it looks like the auto-response is [sent here](https://github.com/FastEndpoints/FastEndpoints/blob/1ba7e459633674ac26fd883bb32b69a2ec7d8057/Src/Library/Endpoint/Endpoint.cs#L98) using the same `CancellationToken` that is passed into the `HandleAsync()` [method call](https://github.com/FastEndpoints/FastEndpoints/blob/1ba7e459633674ac26fd883bb32b69a2ec7d8057/Src/Library/Endpoint/Endpoint.cs#L95) that we we override.

It sends [using the default status code](https://github.com/FastEndpoints/FastEndpoints/blob/1ba7e459633674ac26fd883bb32b69a2ec7d8057/Src/Library/Endpoint/Endpoint.Static.cs#L95C58-L95C77) for the framework's `HttpContext.Response` [which is OK according to docs](https://learn.microsoft.com/en-us/dotnet/api/system.web.httpresponse.statuscode?view=netframework-4.8.1&viewFallbackFrom=net-8.0).

So yes, it has the samantic equivalence to `await SendOkAsync(bibles, ct);` which does an equivalent operation [here](https://github.com/FastEndpoints/FastEndpoints/blob/1ba7e459633674ac26fd883bb32b69a2ec7d8057/Src/Library/Endpoint/Endpoint.Send.cs#L138-L143) under the hood.

🤔 In general I don't like magic properties that do something after a method ends in a framework, especially if I only use them some of the time.  Understanding them requires digging into the framework to understand how it works.  Every developer new to the framework (AKA me) has to break through the nice abstraction and go digging in source code to understand how it works or just assume that it's good without understanding it.

This was hard to discover and I prefer the explicit `await SendOkAsync(bibles, ct);` because it's easier to understand how the response is being sent.

*Is there a strong objection against using the explicit send?*

Philosophy: If you haven't noticed, I tend to weigh readability and understandability higher than fewer keystrokes or lines of code when coding.  Yes, fewer keystrokes and fewer lines of code lends itself to readability but sometimes it obscures what is going on.

Reply:
I don't care either way. I was just pointing it out that it's available in the framework.